### ISSUE TITLE
GPU kernel for segment_sum?

### ISSUE THREAD

**Issue by yycho0108** [tensorflow/tensorflow#11228]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 16.04
- **TensorFlow installed from (source or binary)**: Binary
- **TensorFlow version (use command below)**: ('v1.2.0-5-g435cdfc', '1.2.1')
- **Python version**: 2.7.12
- **CUDA/cuDNN version**: 6.0
- **GPU model and memory**: GTK1070/ 8105MB

^^ I don't think system information is hugely relevant in this case, but writing them down anyways.

### Describe the problem
Describe the problem clearly here. Be sure to convey here why it's a bug in TensorFlow or a feature request.

This is a feature request. I was running *tf.unsorted_segment_sum* on an unsorted list of segments, and thought that I might be able to get a performance gain if the segment ids are sorted (as that is usually the case).
However, to my surprise, I discovered that there are no supported GPU kernels for vanilla segment_sum, forcing tensorflow to copy memory over to the CPU and thereby slowing down the operation greatly.
Would it be possible to support an optimized GPU version of segment_sum that takes advantage of the fact that the segments are sorted?

### Source code / logs

(Device Placement)
UnsortedSegmentSum: (UnsortedSegmentSum)/job:localhost/replica:0/task:0/gpu:0
softmax/ops/SegmentSum: (SegmentSum)/job:localhost/replica:0/task:0/cpu:0

(Profiling)
[Unsorted Segment Sum] : Took 1.979 Seconds
[Segment Sum] : Took 2.704 Seconds

**Comment by tjingrant** [comment]
I'm working on this one.

**Comment by nolanliou** [comment]
@tjingrant How is the PR going?

**Comment by tjingrant** [comment]
Hi, I got some feedback about a week ago but since then was caught up with some other things. I will likely be able to submit an update within 2 days. Sorry for the delay.

**Comment by yongtang** [comment]
It looks like this issue has been fixed in PR #11630. I will close this one. Thanks for the contribution!
